metpy/calc/thermo.py

```diff
@@ -840,3 +842,180 @@ def relative_humidity_from_specific_humidity(specific_humidity, temperature, pre
+                     tmpk.shape[-2], axis=1)[:, :, np.newaxis],
+                     tmpk.shape[-1], axis=2)
+    isentlevs2 = np.repeat(np.repeat(isentlevels[:, np.newaxis],
+                           tmpk.shape[-2], axis=1)[:, :, np.newaxis],
+                           tmpk.shape[-1], axis=2)
+
+    # exponent to Poisson's Equation, which is imported above
+    ka = kappa * 1000 * units('g/kg')
+
+    # calculate theta for each point
+    thtalevs = potential_temperature(levs, tmpk)
+
+    # Find log of pressure to implement assumption of linear temperature dependence on
+    # ln(p)
+    pk = np.log(levs.m)
+
+    # Calculations for interpolation routine
+    pok = 1000. ** (ka)
+
+    # index values for each point for the pressure level nearest to the desired theta level
+    minv = np.apply_along_axis(np.searchsorted, 0, thtalevs.m, isentlevs.m)
+    ar = np.arange
+
+    # calculate constants for the interpolation
+    a = (tmpk.m[minv, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])] -
+         tmpk.m[minv - 1, ar(tmpk.shape[-2]).reshape(-1, 1),
+                ar(tmpk.shape[-1])]) / (pk[minv, ar(tmpk.shape[-2]).reshape(-1, 1),
+                                           ar(tmpk.shape[-1])] -
+                                        pk[minv - 1, ar(tmpk.shape[-2]).reshape(-1, 1),
+                                           ar(tmpk.shape[-1])])
+
+    b = tmpk.m[minv, ar(tmpk.shape[-2]).reshape(-1, 1),
+               ar(tmpk.shape[-1])] - a * pk[minv,
+                                            ar(tmpk.shape[-2]).reshape(-1, 1),
+                                            ar(tmpk.shape[-1])]
+
+    # calculate first guess for interpolation
+    pk1 = (pk[minv, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])] +
+           0.5 * (pk[minv + 1, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])] -
+                  pk[minv, ar(tmpk.shape[-2]).reshape(-1, 1), ar(tmpk.shape[-1])]))
+
+    # iterative interpolation useing scipy.optimize.fixed_point and _isen_iter defined above
+    pk2 = so.fixed_point(_isen_iter, pk1, args=(isentlevs2.m, ka, a, b, pok),
+                         xtol=eps, maxiter=max_iters)
+
+    # get back pressure and assign nan for values with pressure greater than 1000 hPa
+    isentprs = np.exp(pk2)
+    isentprs[isentprs > 1000.] = np.nan
+
+    # create list for storing output data
+    ret = []
+    ret.append(isentprs * units.hPa)
+
+    # if tmpk_out = true, calculate temperature and output as last item in list
+    if tmpk_out:
+        ret.append((isentlevs2.m / ((1000. / isentprs) ** kappa)) * units.kelvin)
+
+    # check to see if any additional arguments were given, if so, interpolate to
+    # new isentropic levels
+    try:
+        args[0]
+    except IndexError:
+        return ret
+    else:
+        # Flip data for pressure in increasing order
+        pk3 = np.flipud(pk2)
+
+        # do an interpolation for each additional argument
+        for i in range(0, len(args)):
```

Comment:
How about:
```python
for arr in args:
```
eliminate `c`, then if necessary do `arr = np.flipud(arr)`. Replace all uses of `d` with `arr`.